tools/c7n_azure/tests/test_vm.py

```diff
@@ -108,12 +105,18 @@ def test_stop(self):
                  'op': 'in',
                  'value_type': 'swap',
                  'value': 'PowerState/running'}],
+            'actions': [
+                {'type': 'start'}
+            ]
         })
-        resources = p.run()
-        self.assertEqual(len(resources), 0)
+        p.run()
+        start_action_mock.assert_called()
```

Comment:
I think there is `assert_called_with(params)` available that lets you specify the params you expect so you know the method is really getting called right.